sensu/sensu-go
# [GraphQL] Queries for 'Event Filter' Resources

## What is this change?

This PR adds the ability to query for Event Filter resources from the GraphQL schema to support https://github.com/sensu/sensu-enterprise-go/issues/501. There are two ways users can retrieve handlers.

Users can retrieve a single Event Filter given a namespace and a name from the `eventFilter` field on the root query type,
Or, users can retrieve a collection of Event Filters (EventFiltersCollection) from the `eventFilters` field on the `namespace` type.

<img width="1027" alt="Screen Shot 2019-06-25 at 1 13 51 PM" src="https://user-images.githubusercontent.com/3856248/60129959-1e683d00-974b-11e9-9b21-8507ac274914.png">


## Why is this change necessary?

This change supports a future update in the in the `sensu/web` repo (https://github.com/sensu/sensu-enterprise-go/issues/507) where Sensu users will have the ability to view a list of event filters and see the details of a particular selected event filter. This change allows the Web UI to query for event filters in order to display this data to the use

## Does your change need a Changelog entry?
Yes,

## How did you verify this change?
To manually verify this change you can run a sensu-backend and query the GraphQL endpoint via GraphiQL.

1. Run a sensu-backend locally
2. Create a number of event filters. I did this using `sensuctl -create --file example_event_filters.json` 
3. Run the sensu/web application locally with the apollo dev tools Chrome extension installed. (Don't forge to update the graphQLSchemaRef in package.json to the latest commit on this PR `76e06c0`)
4. Run queries to ensure that handlers are being returned correctly.
example query:
```gql
query {
  eventFilter(namespace: "sensu-devel", name: "filter-production-only") {
    name
    namespace
    expressions
    runtimeAssets {
      name
    }
  }
  
  namespace(name: "sensu-devel") {
    eventFilters {
      nodes {
        name
      }
    }
  }
}
```
